geomesa-process/geomesa-process-vector/src/main/scala/org/locationtech/geomesa/process/analytic/DensityProcess.scala

```diff
@@ -161,6 +167,9 @@ class DensityProcess extends GeoMesaProcess {
     invertedQuery.getHints.put(QueryHints.DENSITY_BBOX, envelope)
     invertedQuery.getHints.put(QueryHints.DENSITY_WIDTH, outputWidth)
     invertedQuery.getHints.put(QueryHints.DENSITY_HEIGHT, outputHeight)
+    if (argGeomAttr != null) {
```

Comment:
As the query is inverted...  is the bbox applied to the default geometry or the 'desnity' geometry?

Reply:
The envelope is pulled out and applied against the 'density' geometry in `QueryRunner.configureQuery`

Reply:
Should we add a logger.warn in case folks query a non-indexed geometry?  

I'm wondering if we are letting people do something they'll regret...

Reply:
haha, this was your idea...
if you query against a non-indexed geom we already log a warning about full table scans.

Reply:
i think if they're calling it through geoserver there will be a bbox filter on the primary geometry, and then a secondary filter on the second geometry. that may result in some 'missing' data if the primary geom isn't in view but the second one is, but _shrugs_

Reply:
For the literal 'edge' case, I wonder if we need to provide some buffering mechanism?  Hmm...

Reply:
we do buffer the envelope here: https://github.com/locationtech/geomesa/pull/2522/files#diff-37e08ab303976058cacaadf59c39c4c7R157